flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -105,84 +91,38 @@ public KafkaPartitionSplitReader(
     }
 
     @Override
-    public RecordsWithSplitIds<Tuple3<T, Long, Long>> fetch() throws IOException {
-        KafkaPartitionSplitRecords<Tuple3<T, Long, Long>> recordsBySplits =
-                new KafkaPartitionSplitRecords<>();
+    public RecordsWithSplitIds<ConsumerRecord<byte[], byte[]>> fetch() throws IOException {
         ConsumerRecords<byte[], byte[]> consumerRecords;
         try {
             consumerRecords = consumer.poll(Duration.ofMillis(POLL_TIMEOUT));
         } catch (WakeupException we) {
-            recordsBySplits.prepareForRead();
-            return recordsBySplits;
+            return null;
```

Comment:
It seems that rerunning null here still might cause NPE in the `FetchTask.run()` even though null is only returned when the consumer is wakenup.

Reply:
Thanks for the review @becketqin ! It's really a good catch. Fixed in the latest push